### PR TITLE
Support transient storage in optimized state

### PR DESCRIPTION
**Pull request by petertdavies**
### What was wrong?

The optimized state didn't work with Cancun.

### How was it fixed?

Add Cancun support to optimized state.

#### Cute Animal Picture

Put a link to a cute animal picture inside the parenthesis-->

### CODE REVIEW

**Comment by gurukamath** [comment]
Pushed a commit to re-enable the optimised tests

File: src/ethereum/cancun/state.py [hunk 1]
```diff
@@ -98,19 +98,26 @@ def begin_transaction(
         )
 
 
-def commit_transaction(state: State) -> None:
+def commit_transaction(
+    state: State, transient_storage: Optional[TransientStorage] = None
```
Comment (SamWilsn) [inline src/ethereum/cancun/state.py:102]: Does this need to be reworked after #918?

File: src/ethereum_optimized/state_db.py [hunk 2]
```diff
@@ -250,6 +301,30 @@ def rollback_transaction(state: State) -> None:
         if not state.tx_restore_points:
             state.created_accounts.clear()
 
+    if has_transient_storage:
+
+        @add_item(patches)
+        def rollback_transaction(
+            state: State,
+            transient_storage: Optional[Any] = None,
+        ) -> None:
```
Comment (SamWilsn) [inline src/ethereum_optimized/state_db.py:310]: The original function has a signature of:

https://github.com/ethereum/execution-specs/blob/11bb742f063335d1c94878356d02bf49cafb08d4/src/ethereum/cancun/state.py#L121-L123

Do we still need the `Optional` in the patch?

Comment (petertdavies) [reply src/ethereum_optimized/state_db.py:310]: No, but it's harmless.

**Review by SamWilsn — APPROVED** [review]

**Pull request merged**